### PR TITLE
Rewrite environment variable sections of all projects

### PR DESCRIPTION
**Pull request by Stezido**
## Tasks

- [x] Remove unused NODE_ENV in api
- [x] Update .env_example
- [x] Create testcluster script
#### Check and rewrite env table:
- [x] frontend
- [x] api
- [x] blockchain
- [x] provisioning
- [x] excel-export
- [x] email-notification
- [x] Introduction.md
- [x] Upload security FAQ
- [x] Upload a software architecture diagram of Trubudget

### CODE REVIEW

File: blockchain/README.md [hunk 1]
```diff
@@ -26,35 +26,51 @@ As described before: The Trubudget Blockchain is created one by a master, which
...
+| API_PORT                    | no       | 8080          | The port used to connect to the master-node's api.                                                                                                                                                                                                                                                                                                 |
+| API_PROTO                   | no       | http          | The Protoll which should be used to connect to the master-node's api. (http, https)                                                                                                                                                                                                                                                                |
```
Comment (daniel-arnauer) [inline blockchain/README.md:37]: _Protoll_ - Protocoll

File: doc/security/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
+# Security @ Trubudget
+
+This document shows the security level of Trubudget via FAQ. Do not hesitate to ask a security question, so the Trubudget team can answer it and expand the FAQ - just open an issue.
+
+## FAQ
+
+### Why an implementation of TruBudget with Blockchain Multichain technology? What are the reasons or advantages compared to other Blockchain implementations or technologied?
+
+Multichain is a hi-performance private/permissioned blockchain technology which allows high-throughput on standard hardware. Additionally it work well in cases of network partitions.
```
Comment (georgimld) [inline doc/security/README.md:9]: high-performance not hi-performance?

File: blockchain/README.md [hunk 3]
```diff
@@ -26,35 +26,51 @@ As described before: The Trubudget Blockchain is created one by a master, which
...
+| P2P_HOST                    | no       |               | The IP address of the blockchain node you want to connect to. When given, the node joins the existing network rather than creating its own chain.                                                                                                                                                                                                  |
+| P2P_PORT                    | no       | 7447          | The port on which the node you want to connect to has exposed the blockchain.                                                                                                                                                                                                                                                                      |
```
Comment (daniel-arnauer) [inline blockchain/README.md:43]: _The port on which the node you want to connect to has exposed the blockchain._ to
_The port on which the node you want to connect to has to be exposed to the blockchain._

File: blockchain/README.md [hunk 4]
```diff
@@ -26,35 +26,51 @@ As described before: The Trubudget Blockchain is created one by a master, which
...
+| ------------ | -------- | ------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| EXPOSE_MC    | no       | false         | If set to true Trubudget tries to expose it's multichain via kubernetes' ingress. First a kubernetes config file is searched in `/.kube/config`. If not found and the MC is deployed in a cluster it is search for the service account to get the information. After configuration the `externalip` arg is set to the ip of the service' ingress of the configured cluster SERVICE_NAME and NAMESPACE are required to access the ingress of the service. |
```
Comment (daniel-arnauer) [inline blockchain/README.md:71]: If set to true**,** Trubudget tries to expose it's multichain via kubernetes' ingress


If not found and the MC is deployed in a cluster**, it is searching** for the service account to get the information

File: doc/security/README.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
+# Security @ Trubudget
+
+This document shows the security level of Trubudget via FAQ. Do not hesitate to ask a security question, so the Trubudget team can answer it and expand the FAQ - just open an issue.
+
+## FAQ
+
+### Why an implementation of TruBudget with Blockchain Multichain technology? What are the reasons or advantages compared to other Blockchain implementations or technologied?
```
Comment (daniel-arnauer) [inline doc/security/README.md:7]: technologied - technologies

File: doc/security/README.md (hunk 2 above)
Comment (daniel-arnauer) [inline doc/security/README.md:9]: _Additionally it work_ - Additionally, it works

File: doc/security/README.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+#### Examples
+
+- Request to sign a contract by stakeholder A – Upload of signed contract by stakeholder B
+- Request to approve a report by A – Upload of approved report by B
+- Request to give no-objection to tender documents by A– Documented no-objection by B
+- Payment documented by A – Confirmation of receipt by B.
+- Information by A – Validation by B.
+
+Internal processes are not supposed to be part of TruBudget as it is a mulit-stake-holder tool. TruBudget ist very flexible in regard to the applied processes. In each use case the stakeholders will decide on relevant processes to be put on TruBudget in order to facilitate their joint business.
```
Comment (daniel-arnauer) [inline doc/security/README.md:31]: TruBudget ist - TruBudget is

File: doc/security/README.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+### Is data confidentiality taken into account in TruBudget? If yes, how?
+
+Data confidentiality is guaranteed on API level. Users need to be authenticated and authorized in order to request data from the API based on a role-base access model. Data at rest (in the blockchain) is visible to persons with access to production server.
+
+### In terms of application security, does TruBudget integrate the concept of multi- signature?
+
+No. Transactions are signed with the user’s private key which is related to an organi-za¬tional wide secret. Transactions can only be signed on nodes of the user’s organization. Multiple signing is possible, but not implemented (e.g. 4 eyes principle). However, within an organization writing rights on TruBudget may be organized according to internal requirements (e.g. internal approvals before entering data).
```
Comment (daniel-arnauer) [inline doc/security/README.md:39]: organi-za¬tional ?

File: doc/Introduction.md [hunk 8]
```diff
@@ -24,30 +32,61 @@ echo %VARIABLE% (Windows)
...
+| ------------ | -------- | ------------- | ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| EXPOSE_MC    | no       | false         | bc      | If set to true Trubudget tries to expose it's multichain via kubernetes' ingress. First a kubernetes config file is searched in `/.kube/config`. If not found and the MC is deployed in a cluster it is search for the service account to get the information. After configuration the `externalip` arg is set to the ip of the service' ingress of the configured clusterSERVICE_NAME and NAMESPACE are required to access the ingress of the service. |
```
Comment (daniel-arnauer) [inline doc/Introduction.md:87]: If not found and the MC is deployed in a cluster**, that it is searching** for the service account to get the information.

File: excel-export/README.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+- 1 Excel-export service
+
+To check what is configurable regarding excel-export service check out the [environment variables section](#environment-variables)
+
+### Endpoints
+
+| Method | Endpoint   | Description                                                              |
+| ------ | ---------- | ------------------------------------------------------------------------ |
+| GET    | /health    | Checks if email service is up                                            |
+| GET    | /readiness | Checks if email service is ready                                         |
```
Comment (daniel-arnauer) [inline excel-export/README.md:43]: **excel** not **email**

File: excel-export/README.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+For details see [Blockchain environment variables](../blockchain/README.md#email-service).
+
+### Frontend
+
+For details see [Blockchain environment variables](../frontend/README.md#email-service).
+
+## Getting Started
+
+The easiest way to get started is to use our pre-set `docker-compose` cluster which starts the whole TruBudget application including all email components (that means you need to install [Docker](https://www.docker.com/community-edition#/download)).
```
Comment (daniel-arnauer) [inline excel-export/README.md:28]: excel not email

File: excel-export/README.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+| ACCESS_CONTROL_ALLOW_ORIGIN | "\*"          | Since the export service uses CORS, the domain by which it can be called needs to be set. Setting this value to `"*"` means that it can be called from any domain. Read more about this topic [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS). |
+
+### Blockchain
+
+For details see [Blockchain environment variables](../blockchain/README.md#email-service).
+
+### Frontend
+
+For details see [Blockchain environment variables](../frontend/README.md#email-service).
+
```
Comment (daniel-arnauer) [inline excel-export/README.md:25]: i thing the link does net `#email-service`

File: frontend/README.md [hunk 12]
```diff
@@ -2,6 +2,34 @@
...
+| REACT_APP_VERSION | -             | Injected version via `$npm_package_version` in`.env` file to ensure the version is shown in the frontend                                                             |
+| PROD_API_HOST     | -             | IP address of the api with production environment. This is only required if nginx proxy is used <br>**Hint:** When deployed locally the host is set to localhost     |
+| PROD_API_PORT     | 8080          | Port of the api with production environment This is only required if nginx proxy is used                                                                             |
+| TEST_API_HOST     | -             | IP address of the api with test environment This is only required if nginx proxy is used. <br>**Hint:** When deployed locally the host is set to localhost           |
+| TEST_API_PORT     | 8080          | Port of the api with test environment This is only required if nginx proxy is used                                                                                   |
```
Comment (daniel-arnauer) [inline frontend/README.md:16]: **.** This

File: blockchain/README.md (hunk 3 above)
Comment (Stezido) [reply blockchain/README.md:43]: This should be right. The blockchain has to be exposed via P2P_PORT. Your sentence describe the port has to be exposed.

**Review by georgimld — APPROVED** [review]

**Pull request merged**